include/openssl/stack.h

```diff
@@ -492,12 +495,29 @@ BSSL_NAMESPACE_END
                          (OPENSSL_sk_delete_if_func)func, data);               \
   }                                                                            \
                                                                                \
-  OPENSSL_INLINE int sk_##name##_find(const STACK_OF(name) *sk,                \
+  /* use 3-arg sk_*_find_awslc when |out_index| needed */                      \
+  OPENSSL_INLINE int sk_##name##_find_awslc(const STACK_OF(name) *sk,          \
                                       size_t *out_index, constptrtype p) {     \
     return OPENSSL_sk_find((const OPENSSL_STACK *)sk, out_index,               \
                            (const void *)p, sk_##name##_call_cmp_func);        \
   }                                                                            \
                                                                                \
+  /* use 2-arg sk_*_find for OpenSSL compatibility */                          \
+  OPENSSL_INLINE int sk_##name##_find(const STACK_OF(name) *sk,                \
+                                      constptrtype p) {                        \
+    const size_t mask = sizeof(size_t) > sizeof(int)                           \
```

Comment:
The mask should also cover the sign bit, even if `sizeof(size_t) == sizeof(int)`:
```
const size_t mask = ~((size_t) 0)) << (sizeof(int) * 8 - 1);
```

I don't think it's the case for any supported platform that 
`sizeof(size_t) < sizeof(int)`

Reply:
> The mask should also cover the sign bit, even if `sizeof(size_t) == sizeof(int)`

`size_t` is always unsigned, no?

> I don't think it's the case for any supported platform that `sizeof(size_t) < sizeof(int)`

I didn't think so either, but am concerned with the `sizeof(size_t) == sizeof(int)` case, hence the hard inequality. Initially I'd written this as `sizeof(size_t) != sizeof(int)`, used `<` to guard against cases I hadn't considered.

In any case, something is wrong with my implementation here. The [`ubuntu1604_gcc5x_x86` test is failing](https://tiny.amazon.com/zhmkmmgs/IsenLink) with a compile error:

```
../include/openssl/stack.h:509:46: error: left shift count >= width of type [-Werror=shift-count-overflow]
         ? (~((size_t) 0)) << (sizeof(int) * 8)                                 \
                                              ^
```

Same with `centos7_gcc4x_x86_64_fips`. I'll look into how earlier versions of GCC handle bit shifting.

Reply:
I ended up dropping the masking logic altogether in favor of checking `INT_MAX` and casting.